functions/New-DbaLinkedServerLogin.ps1

```diff
@@ -0,0 +1,135 @@
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Security, Server
+        Author: Adam Lancaster https://github.com/lancasteradam
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaLinkedServerLogin
+
+    .EXAMPLE
+        PS C:\>New-DbaLinkedServerLogin -SqlInstance sql01 -LinkedServer linkedServer1 -LocalLogin localUser1 -RemoteUser remoteUser1 -RemoteUserPassword <password>
+
+        Creates a new linked server login and maps the local login testuser1 to the remote login testuser2. This linked server login is created on the sql01 instance for the linkedServer1 linked server.
+
+        NOTE: passwords are sent to the SQL Server instance in plain text. Check with your security administrator before using the command with the RemoteUserPassword parameter. View the documentation for sp_addlinkedsrvlogin for more details.
+
+    .EXAMPLE
+        PS C:\>New-DbaLinkedServerLogin -SqlInstance sql01 -LinkedServer linkedServer1 -Impersonate
+
+        Creates a mapping for all local logins on sql01 to connect using their own credentials to the linked server linkedServer1.
+
+    .EXAMPLE
+        PS C:\>Get-DbaLinkedServer -SqlInstance sql01 -LinkedServer linkedServer1 | New-DbaLinkedServerLogin -LinkedServer linkedServer1 -LocalLogin testuser1 -RemoteUser testuser2 -RemoteUserPassword <password>
+
+        Creates a new linked server login and maps the local login testuser1 to the remote login testuser2. This linked server login is created on the sql01 instance for the linkedServer1 linked server. The linkedServer1 instance is passed in via pipeline.
+
+        NOTE: passwords are sent to the SQL Server instance in plain text. Check with your security administrator before using the command with the RemoteUserPassword parameter. View the documentation for sp_addlinkedsrvlogin for more details.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$LinkedServer,
+        [string]$LocalLogin,
+        [string]$RemoteUser,
+        [Security.SecureString]$RemoteUserPassword,
+        [switch]$Impersonate,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.LinkedServer[]]$InputObject,
+        [switch]$EnableException
+    )
+    process {
+
+        foreach ($instance in $SqlInstance) {
+
+            if (Test-Bound -Not -ParameterName LinkedServer) {
+                Stop-Function -Message "LinkedServer is required"
```

Comment:
Thank you, Adam! This should go before the foreach so that it only processes once.